saleor/graphql/core/validators.py

```diff
@@ -10,3 +18,21 @@ def validate_one_of_args_is_in_query(arg1_name, arg1, arg2_name, arg2):
         raise GraphQLError(
             f"Either '{arg1_name}'  or '{arg2_name}' argument is required"
         )
+
+
+def validate_price_amount(value: Optional["Decimal"], currency: str = None):
+    """Validate if price amount does not have too many decimal places.
+
+    Price amount can't have more decimal places than currency allow to.
+    """
+
+    # check no needed when there is no value
+    if not value:
+        return
+
+    currency_fraction = get_currency_fraction(currency or settings.DEFAULT_CURRENCY)
+    value = value.normalize()
```

Comment:
Do we need to have a validation function and raise the error?  I am not sure of this. 
If someone passes the 4 numbers after a comma and provides it with USD, then he should be aware that we will clean it up and convert it to 2 places. For me, it is more expected behavior that raising the error because someone passed too many 0 after the comma

Other example. Here we should just clean the price and store the expected numbers after comma.
```
get_currency_fraction("usd")
Out[14]: 2
d = Decimal(11.120000000000)
d = d.normalize()
abs(d.as_tuple().exponent)
Out[20]: 26
```

Reply:
Your flow works only if you create Decimal from str. Decimal("11.10").

Reply:
@patrys suggested to raising an error.

Reply:
The validators will be used only in mutations, and there you get prices in format Decimal("11.10").

Reply:
Yes, it is used in mutation only for now, but IMO if it doesn't work with the Decimal created from a number we should at least add  such info to docstring

Reply:
Ok, I added info to the docstring.

Reply:
I suggested raising an error when we would lose information. Examples:

* Passing 1.00 as USD is fine
* Passing 1 as USD is fine
* Passing 1.23000000 as USD is fine (all trailing digits are zeroes)
* Passing 1.001 as USD is en error (we would lose the trailing one)